lib/src/shared/message.dart

```diff
@@ -54,25 +57,33 @@ class GrpcMessageSink extends Sink<GrpcMessage> {
   }
 }
 
-List<int> frame(List<int> payload) {
-  final payloadLength = payload.length;
+List<int> frame(List<int> rawPayload, [Codec codec]) {
+  final compressedPayload =
+      codec == null ? rawPayload : codec.compress(rawPayload);
+  final payloadLength = compressedPayload.length;
   final bytes = Uint8List(payloadLength + 5);
   final header = bytes.buffer.asByteData(0, 5);
-  header.setUint8(0, 0); // TODO(dart-lang/grpc-dart#6): Handle compression
+  header.setUint8(0, codec == null ? 0 : 1);
```

Comment:
we can probably avoid sending `1` here if codec is identity codec?

Reply:
We shouldn't do this since the headers would be reporting the `grpc-encoding` and would just cause additional checks for no meaningful gains.

Reply:
Makes sense, just wanted to check.